xarray/test/test_dataarray.py

```diff
@@ -1407,6 +1407,32 @@ def test_resample_first(self):
                              name='time')
         self.assertDataArrayIdentical(expected, actual)
 
+    def test_resample_first_keep_attrs(self):
+        times = pd.date_range('2000-01-01', freq='6H', periods=10)
+        array = DataArray(np.arange(10), [('time', times)])
+        array.attrs['meta'] = 'data'
```

Comment:
Readability still counts in test code. I find unit tests that copy all but a line or two from a related test difficult to distinguish between (e.g. `test_resample_first_keep_attrs` vs. `test_resample_mean_keep_attrs`).

  I don't entirely buy the SO post's perspective but he does contradict your point to some extent:

> If you can factor something in test code and you're sure it's ok, then do it.